### ISSUE TITLE
pre-train model

### ISSUE THREAD

**Issue by linchaozhang** [farmingyard/ShuffleNet#1]
Have you successfully trained the model?
can you sent me a pre-train model, i want to train a shufflenet-ssd with a pre-train model.

thanks!

**Comment by farmingyard** [comment]
@linchaozhang The model is training right now.

**Comment by KeyKy** [comment]
@farmingyard what is your currency accuracy of val?

**Comment by leolee96** [comment]
Hi @farmingyard ,
I'm sort of confused about your stride case in your deploy prototxt. Why you don't shuffle the channel here?
thanks!

**Comment by farmingyard** [comment]
@leolee96 Do you mean the stride=2 case? Yeah, its my mistake, i have fixed it. Thanks.

**Comment by ananddb90** [comment]
+1 for pre trained model
@farmingyard  thank you for your share :)

**Comment by ananddb90** [comment]
@farmingyard is it possible for you to share solver file also ?:=)

**Comment by suzhenghang** [comment]
Linear lr seems to work well

**Comment by farmingyard** [comment]
@linchaozhang @KeyKy @leolee96 @ananddb90 
The pre-trained model is released, please check ReadMe file.

**Comment by KeyKy** [comment]
THX!

**Comment by linchaozhang** [comment]
@farmingyard 
Thanks a lot.
But i found some differences between prototxt and caffemodel, in 'resx5_conv1', num_output : 60 in your prototxt, and 240 in your model. i think it should be 60 for 'resx5_conv1'(because the number of
bottleneck channels to 1/4 of the output channels for each ShuffleNet unit, said in paper)

**Comment by farmingyard** [comment]
@linchaozhang 
I think you may had a mistake，cause the model is trained with the deploy prototxt file.
BTW，how can you get the model's layer output？

**Comment by linchaozhang** [comment]
@farmingyard 
I have found the real difference, missing the param "group" in resx5_conv1 and resx13_conv1 in your prototxt, and missing the shuffle_layer, i think you forget to update the prototxt.

**Comment by farmingyard** [comment]
@linchaozhang 
Sorry for that I updated  the deploy file 9 days ago, you may not follow it. Please check the update info.
https://github.com/farmingyard/ShuffleNet/commit/fdb6e836072581cd465de2d9d07bc6571be86fca

**Comment by 123chengbo** [comment]
i use your model and deploy.prototxt for imagenet classification， but result seems to be totally wrong（result labels are all wrong）， i compare it to caffenet model、deploy.prototxt（result labels are right）。

**Comment by xchani** [comment]
Hi, @farmingyard 
Did you do any data augmentation in training stage?

**Comment by farmingyard** [comment]
@xchani 
Just crop and mirror.

**Comment by fujihuang** [comment]
@farmingyard  Hi, thanks for your sharing, great job. And i just wonder where is your pre-trained model, you have mentioned that have released, but i just can not see anything~~I hope i have not missed anything

**Comment by yunren** [comment]
@farmingyard 
Hi, thanks for sharing  your great job.
 And I just wonder where is your pre-trained model that you have mentioned above, but i just can not find it after having checked ReadMe file

**Comment by farmingyard** [comment]
@fujihuang @yunren
Sorry for some reasons that I can't release the pre-trained model any more.

**Comment by CrazySssst** [comment]
any one can send the pre-trained model to me ? i really need it, thanks for your help   @linchaozhang @farmingyard @KeyKy @leolee96 @ananddb90

**Comment by ananddb90** [comment]
can you send me your mail-id

**Comment by yonghenglh6** [comment]
Hi, would u put up the val accuracy of your model, please?

**Comment by dingdashen** [comment]
Hi,ananddb90
 can you send the pre-trained model to me? I want finetune it to VOC  dataset.

**Comment by CrazySssst** [comment]
@ananddb90   this is my email : push_back_max@163.com , thank you !

**Comment by acdreamdpl** [comment]
any one can send the pre-trained model to me? I hope to finetune it on faster-rcnn. @farmingyard @linchaozhang @KeyKy @leolee96 @ananddb90 .  thanks

**Comment by ananddb90** [comment]
https://bitbucket.org/ananddb90/shuffle_net/src

please share your experience after fine tunning on detection problem as I am also trying the same and yet not able to get good results

**Comment by CrazySssst** [comment]
thanks for you sharing.
i will share my results on SSD after finish my experiment

**Comment by 7oud** [comment]
@ananddb90 
Thanks for your sharing.
Can you put up the accuracy of your sharing ShuffleNet on ImageNet？

**Comment by zoufangyu1987** [comment]
@farmingyard 
@ananddb90 
@linchaozhang 
@KeyKy
@leolee96 
Do you have 0.5_shufflenet and 0.75_shufflenet caffe  pre-trained model，train.prototxt?
thank you !

**Comment by zoufangyu1987** [comment]
@farmingyard @ananddb90 @linchaozhang @KeyKy @leolee96
I need to run ShuffleNet on "ARM" or "DSP",
so that speed is the most important.
Do you have any suggestion?
Thank you very much !

**Comment by felixfuu** [comment]
@123chengbo I have the same problem with you , do you find the cause of this problem?Looking forward to your reply!

**Comment by farmingyard** [comment]
@ananddb90 
I also found that the shufflenet with ssd is not good. sad!

**Comment by felixfuu** [comment]
@farmingyard i use the caffemodel @ananddb90 provided，but result seems to be totally wrong（on ImageNet 2012）

**Comment by farmingyard** [comment]
@felixfuu 
It maybe your test prototxt file is not same with @ananddb90 's caffemodel, please double check.

**Comment by farmingyard** [comment]
@zoufangyu1987 
You can use ncnn's arm neon codes for 3x3 conv, it is more fast, for 1x1 conv, you can use eigen's gemm, for 3x3 depthwise conv, you can modify ncnn's 3x3 conv codes.

**Comment by felixfuu** [comment]
@farmingyard i use the deploy.prototxt provided by you

**Comment by zoufangyu1987** [comment]
@farmingyard 
Thank you very much!
What is ncnn`s and eigen`s network link address, please?

**Comment by felixfuu** [comment]
@ananddb90 Can you share the test.prototxt ?

**Comment by ananddb90** [comment]
sorry guys, I don't have  0.5_shufflenet and 0.75_shufflenet caffe pre-trained model/prototxt
@felixfuu I used files from this repository 
@farmingyard  yes, I also got bad results for ssd and even worse for RFCN.

**Comment by farmingyard** [comment]
@zoufangyu1987 
ncnn:https://github.com/Tencent/ncnn
eigen:http://eigen.tuxfamily.org/index.php?title=Main_Page

**Comment by farmingyard** [comment]
@felixfuu 
Well, if the prototxt file is same to caffemodel and the result is not correct, the possible reason may be your incorrect label or different data preprocess, you can check!

**Comment by zoufangyu1987** [comment]
@farmingyard 
THANK YOU !

**Comment by felixfuu** [comment]
@farmingyard @ananddb90 Thank you~

**Comment by MissDores** [comment]
Dear friend, could you help send your pre-trained models to my email box, 13261520092@163.com? Thank you very much!  I am also trying to implement detection task based on ShuffleNet and FRCNN. Have you achieved ideal results? @ananddb90

**Comment by zoufangyu1987** [comment]
@MissDores 
You can download shufflenet.caffemodel from this network.
https://bitbucket.org/ananddb90/shuffle_net/src 
Thank @ananddb90!

**Comment by MissDores** [comment]
@zoufangyu1987 @ananddb90 
Thank you so much!

**Comment by nnop** [comment]
What was performance when you train the model? @ananddb90

**Comment by l1a0t1n9** [comment]
Hi,ananddb90, can you send the pre-trained model to me? liaoting@whu.edu.cn Thank you very much!@ananddb90

**Comment by ryancuicui** [comment]
@ananddb90  HI， i opened the website you provided,but i can not access the pre-train caffemodel.Can you provided the caffemodel again? THANKS!!!

**Comment by colinwubot** [comment]
Hi friend, your works is wonderful. Could you share your pre-trained models with my? Thank you very much!  @ananddb90

**Comment by tzt101** [comment]
@ananddb90 Hi, Could you share your pre-trained models? tzt101@outlook.com Thank you very much!

**Comment by dlyldxwl** [comment]
@ananddb90 @zoufangyu1987  Thanks  for your sharing ! But the link isn't useful, can you share the caffemodel again or send to me ? My email is 844518133@qq.com .
Thanks again！

**Comment by mameng1** [comment]
@ananddb90 @zoufangyu1987  Thanks for your sharing ! But the link isn't useful, can you share the caffemodel again or send to me ? My email is mameng2908071445@gmail.com .
Thanks again！

**Comment by ananddb90** [comment]
hello all,
apologies for the late reply. Also I am sorry to say that I am not allowed to make it public yet 
I will write down once it will be public again :)

**Comment by zoufangyu1987** [comment]
@dlyldxwl @mameng1  
I'm sorry to have kept you waiting.
You can download it by "https://pan.baidu.com/s/1eS8NOm2", password is  "bcj6".

**Comment by dlyldxwl** [comment]
OK, i get it。
thanks again ~

**Comment by zoufangyu1987** [comment]
@dlyldxwl 
You are welcome！

**Comment by mydear33000** [comment]
@zoufangyu1987 
Thanks a lot!

**Comment by fujiaweigege** [comment]
Could anyone share train_val.prototxt and solver.prototxt of ShuffleNet, thanks a lot!

**Comment by dlyldxwl** [comment]
@fujiaweigege I think the origin deploy.prototxt of shufflenet has some issue in scale, I add pad param in some layers. i retain the  shufflenet_ssd train.prototxt,becase of network connection ,I can't upload file,you can send me your email.Of course I don’t even make sure it’s much accurate.

**Comment by fujiaweigege** [comment]
@dlyldxwl Thank you very much!

**Comment by Justin-Yuan** [comment]
great work!! I am working on a project on human-tracking robots with microcontroller and would like to try out the pretrained shufflenet. Can you send me the caffemodel and prototxt files. justin.yuan@mail.utoronto.ca. Thank you so much !

**Comment by cjerry1243** [comment]
The link is now broken.
Could any one send the pretrained model to me?
cjerry1243@gmail.com

**Comment by duan348733684** [comment]
@dlyldxwl   can you  send  your  sufflenet_ssd train.prototxt to my Email 348733684@qq.com ? thanks

**Comment by neelakanth** [comment]
@zoufangyu1987 I am not able to download  "https://pan.baidu.com/s/1eS8NOm2". Link gets reset after every few KBs of download. Can you upload on dropbox or googledrive

If any of you have pre-trained model, corresponding proto, top1/5 error and test image list please upload.

**Comment by zoufangyu1987** [comment]
@neelakanth ,
How much is your e-mail address? I'll send it to you

**Comment by zoufangyu1987** [comment]
My email address is 305434863@qq.com

**Comment by neelakanth** [comment]
@zoufangyu1987 Thanks. Here is my email id neelakanth.shigihalli@gmail.com

**Comment by neelakanth** [comment]
@zoufangyu1987 I am able to use caffemodel that you sent. I took latest code from https://github.com/BVLC/caffe.git and copied shuffle layer code and depth wise convolution code from (https://github.com/farmingyard/caffe-mobilenet)
I tested for 50000 images of imagenet on NVIDIA GPU. I am seeing Top1 62.3% and Top5 84.2%. What Top1 and Top5 you got?

I see that @msnqqer (https://github.com/msnqqer/ShuffleNet) got 65.45%/86.38%. I haven't tried with msnqqer's caffe model yet.

**Comment by zoufangyu1987** [comment]
@neelakanth  Sorry, I didn't train on imagenet data.
I only trained on my own dataset.
So I have no imagenet result.

**Comment by Changkou** [comment]
@ananddb90 Could you please share the pre-training model to my email 383457094@qq.com？ As 
the Bitbucket declined my request. Thanks!!!
